scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/service/ScheduleManager.java

```diff
@@ -226,6 +227,186 @@ private void validateSpecificDateSchedules(List<SpecificDateScheduleEntity> spec
+		boolean isValid = true;
+
+		if (!validateDayOfWeekOrMonth(scheduleBeingProcessed, recurringSchedule)) {
+			isValid = false;
+		}
+
+		if (!validateStartEndDate(scheduleBeingProcessed, recurringSchedule)) {
+			isValid = false;
+		}
+
+		if (!validateStartEndTime(scheduleBeingProcessed, recurringSchedule)) {
+			isValid = false;
+		}
+
+		RecurringScheduleTime time = null;
+		if (isValid) {
+			time = new RecurringScheduleTime(scheduleBeingProcessed, recurringSchedule);
+		}
+
+		return time;
+	}
+
+	private boolean validateStartEndTime(String scheduleBeingProcessed, RecurringScheduleEntity recurringSchedule) {
+		boolean isValid = true;
+		Date startTime = recurringSchedule.getStartTime();
+		Date endTime = recurringSchedule.getEndTime();
+
+		if (!DataValidationHelper.isNotNull(startTime)) {
+			isValid = false;
+			validationErrorResult.addFieldError(recurringSchedule, "schedule.data.value.not.specified",
+					scheduleBeingProcessed, "start_time");
+		}
+
+		if (!DataValidationHelper.isNotNull(endTime)) {
+			isValid = false;
+			validationErrorResult.addFieldError(recurringSchedule, "schedule.data.value.not.specified",
+					scheduleBeingProcessed, "end_time");
+		}
+
+		if (isValid) {
+			// If end date time is not after start date time, then dates invalid
+			if (!DataValidationHelper.isAfter(endTime, startTime)) {
+				isValid = false;
+				validationErrorResult.addFieldError(recurringSchedule, "schedule.date.invalid.start.after.end",
+						scheduleBeingProcessed, "end_time", DateHelper.convertTimeToString(endTime), "start_time",
+						DateHelper.convertTimeToString(startTime));
+			}
+		}
+		return isValid;
+	}
+
+	private boolean validateStartEndDate(String scheduleBeingProcessed, RecurringScheduleEntity recurringSchedule) {
+		// Note: For recurring schedule, start and end date are optional so not checking for null
+		boolean isValid = true;
+		Date startDate = recurringSchedule.getStartDate();
+		Date endDate = recurringSchedule.getEndDate();
+		TimeZone timeZone = TimeZone.getTimeZone(recurringSchedule.getTimeZone());
+
+		if (startDate != null) {
+			// it should be after current date.
+			if (!DataValidationHelper.isDateAfterOrEqualsNow(startDate, timeZone)) {
+				isValid = false;
+				validationErrorResult.addFieldError(recurringSchedule, "schedule.date.invalid.before.current",
+						scheduleBeingProcessed, "start_date", DateHelper.convertDateToString(startDate));
+			}
+		}
+
+		if (endDate != null) {
+			// it should be after current date.
+			if (!DataValidationHelper.isDateAfterOrEqualsNow(endDate, timeZone)) {
+				isValid = false;
+				validationErrorResult.addFieldError(recurringSchedule, "schedule.date.invalid.before.current",
+						scheduleBeingProcessed, "end_date", DateHelper.convertDateToString(endDate));
+			}
+		}
+
+		if (startDate != null && endDate != null && isValid) {
+			// Is it ok to have startDate= endDate? Does not make sense in case of recurring schedules
```

Comment:
How/where are we tracking this question?
Why doesn't it make sense if I only need something to occur on a specific date once a year?